spec/function.dd

```diff
@@ -795,18 +795,22 @@ class Bar : Foo
 ------
 )
 
-        $(P Virtual functions all have a hidden parameter called the
-        $(I this) reference, which refers to the class object for which
-        the function is called.
+        $(P 
+            Non-static (i.e., virtual functions and $(D final) member
+            functions) all have a hidden parameter called the
+            $(I this) reference, which refers to the class object for which
+            the function is called.
         )
 
         $(P
-            Functions with `Objective-C` linkage has an additional hidden,
+            Functions with `Objective-C` linkage have an additional hidden,
             unnamed, parameter which is the selector it was called with.
         )
 
         $(P To avoid dynamic binding on member function call, insert
-        base class name before the member function name. For example:
+        base class name before the member function name. Note that when
+        when invoked in this way, the member function will not be passed
+        the hidden $(I this) parameter. For example:
```

Comment:
Are you sure this is the case?
given 

```d
class A { 
int a;
   void foo() {
       writeln(a);
   }; 
}

class B : A { override void foo() {}}
...
auto b = new B;
b.A.foo
```

A.foo definitely needs `this`, it will be adjusted yes, but its still needed.

Reply:
Interesting, as calling `A.foo()` statically does not add `this` and leads to compilation error (not sure what you mean by "it will be adjusted" ?) whereas `a.A.foo()` does add the hidden reference. Feel free to omit this hunk, or, one could clarify with example and counterexample.

Reply:
I think its better to omit it.